### DISCUSSION TITLE
i am doing osmosis hooks and strangelove PFM consumer side

### DISCUSSION THREAD

**Discussion by dzmitry-lahoda** [informalsystems/ibc-rs#1001] [Q&A] [answered]
here https://github.com/ComposableFi/ibc-apps-more-rs

would types (as separate) crate accepted here or it should be sepate repo?

PFM was hack, but it runs even on Cosmos Hub (and dozen other chain)

Hooks runs on dozen chains, including Axelar

**Comment by dzmitry-lahoda** [comment]
hooks and PFM has really different design to https://github.com/cosmos/ibc-rs/pull/719 or ICS hooks based on 719. so they are different.

we want to reuse these in Solana too

**Comment by Farhad-Shabani** [comment]
Regarding the domain types, of course. We intend to keep the `ibc-rs` as the canonical location for various IBC types. Thus totally cool with defining the peer of their proto types in `ibc-proto-rs` as a fresh crate in `ibc-rs`, of course it should be in line with the existing `*-types` crates. 
As for the business logic implementation built on top of the types' crate (e.g. `ibc-app-transfer` on top of the `ibc-app-transfer-types`), it is up to you if you'd like to keep it in a separate repository. Actually it seems more reasonable, especially if your logic might contain some customization for your particular use case rather than being a general-purpose IBC application.

**Reply by dzmitry-lahoda** [reply]
Strangelov pfm and osmosis hook are deployed to dozens ibc chains.

**Reply by Farhad-Shabani** [reply]
We can begin with housing the types. As going forward, we can obtain more details on the scope of the apps' implementation as well. Hopefully be able to accommodate them too.
Just as a heads-up, if they are urgent for you, I doubt we be able to review their logic implementation before Q2 next year.

**Reply by dzmitry-lahoda** [reply]
i have only types. i produce and receive pfm/hooks in cosmwasm(like osmosis and neutron doing). while composable-ibc and composable parachain has its own type and has app impls,  i assume better to share in one location which can be adopted by all cosmwasm contracts and ibc hosts.

**Reply by Farhad-Shabani** [reply, marked as answer]
Okay. That's fine then.
